docusaurus/docs/tutorials/ui-extensions/register-an-extension.md

```diff
@@ -0,0 +1,235 @@
+- **Use the enum for Grafana extension point ids** - if you are registering a component to one of the available Grafana extension points, make sure that you use the [`PluginExtensionPoints` enum exposed by the `@grafana/data`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/pluginExtensions.ts#L121) package.
+
+### Register a component
+
+```tsx
+import { PluginExtensionPoints } from '@grafana/data';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => <div>This is a new tab on the user profile page.</div>,
+});
+```
+
+### Accessing plugin meta in a component
+
+You can use the `usePluginContext()` hook to access any plugin specific meta information inside your component. The hook [returns a `PluginMeta` object](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/plugin.ts#L62). (This can be useful because the component that you register from your plugin won't be rendered under the React tree of your plugin, but somewhere else in the UI.)
+
+```tsx
+import { usePluginContext, PluginExtensionPoints } from '@grafana/data';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => {
+    const { meta } = usePluginContext();
+
+    // The `jsonData` property is an object that your plugin can manage
+    // using the Grafana Rest APIs
+    return <div>Plugin specific setting: {meta.jsonData.foo}</div>;
+  },
+});
+```
+
+### Access plugin state in a component
+
+```tsx
+import { PluginExtensionPoints } from '@grafana/data';
+import { MyCustomDataProvider } from './MyCustomDataProvider';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => (
+    <MyCustomDataProvider>
+      <div>Plugin specific setting: {meta.jsonData.foo}</div>
+    </MyCustomDataProvider>
+  ),
+});
+```
+
+### Hide a component in certain conditions
+
+This is nothing new for developers working with React, just return `null` from your component in order to not render anything (to hide it).
+
+```tsx
+import { usePluginContext, PluginExtensionPoints } from '@grafana/data';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => {
+    const { meta } = usePluginContext();
+
+    // For the sake of the example this condition is relying on a `jsonData` property
+    // that is managed by your plugin
+    if (!meta.jsonData.isExtensionEnabled) {
+      return null;
+    }
+
+    return <div>Plugin specific setting: {meta.jsonData.foo}</div>;
+  },
+});
+```
+
+## Troubleshooting
```

Comment:
Maybe better as a new separate page

Reply:
Yeah, maybe that's a good idea!

Reply:
Will be handled in a follow up pr. Task added in https://github.com/grafana/grafana-community-team/issues/131.